web/modules/custom/rdf_entity_provenance/rdf_entity_provenance.module

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/**
+ * @file
+ * Contains hooks and functions for the rdf_entity_provenance module.
+ */
+
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Field\BaseFieldDefinition;
+
+/**
+ * Implements hook_entity_bundle_field_info().
+ */
+function rdf_entity_provenance_entity_bundle_field_info(EntityTypeInterface $entity_type, $bundle, array $base_field_definitions) {
+  if ($entity_type->id() !== 'rdf_entity' || $bundle !== 'provenance_activity') {
+    return [];
+  }
+
+  $fields = [];
+
+  // The time that the activity started to federate the remote entity. This is
+  // different from the created date as it can be updated if the entity is
+  // federated again.
+  // @see: https://www.w3.org/TR/2013/REC-prov-o-20130430/#startedAtTime
+  $fields['provenance_started'] = BaseFieldDefinition::create('timestamp')
+    ->setLabel(t('Provenance - Created'))
+    ->setDescription(t('The time that the provenance entity was created.'))
+    ->setComputed(FALSE)
+    ->setCardinality(1)
+    ->setRequired(FALSE);
+
+  // The source of the entity is a link to the Agent. However, an Agent property
+  // is not strictly needed so this will be a link towards the source of the
```

Comment:
The thing here is that the Agent is a good way to track the source of information. However, in our current implementation, the source is not  part of our pipeline epic ticket.
As we discussed yesterday during the standup (invalid information for github), for tracking new entities, it would be nice to have the source tracked. This field is meant to achieve just that.

The provenance entities, if you follow the link in the ticket and the attached images, can come in many ways and patterns. We picked the simplest one and even more we stripped off properties that are not necessary. The provenance entities after all are only for our internal use.
Sander's suggestion from yesterday, to hardcode a uri in the pipeline (e.g. the uri of the ctt repo) seems like a solution that can help us with that.